Framework/CurveFitting/src/Algorithms/VesuvioCalculateGammaBackground.cpp

```diff
@@ -542,26 +540,30 @@ void VesuvioCalculateGammaBackground::cacheInstrumentGeometry() {
   m_foils1.resize(nfoils);
   for (size_t i = 0; i < nfoils; ++i) {
     const auto &foil0 = foils0[i];
+    if (!foil0)
+      throw std::invalid_argument("Error - foil component is NULL.");
```

Comment:
In `ConverrtToYSpace::getComponentParameter`, I was forced to pass an `IComponent` rather than `IComponent_const_sptr` to pass a detector to the method. There used to be null pointer check in this method, but as that could no longer be done there, I moved that check to be done outside the method on a detector before it is passed in.

It is an imperfect solution, so let me know if there are any better ways of going about this.